jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpField.java

```diff
@@ -38,50 +42,85 @@ public class HttpField
      */
     public static final QuotedStringTokenizer NAME_VALUE_TOKENIZER = QuotedStringTokenizer.builder().delimiters("=").build();
 
-    private static final String __zeroQuality = "q=0";
+    private static final String ZERO_QUALITY = "q=0";
+
     private final HttpHeader _header;
     private final String _name;
     private final String _value;
     private int _hash = 0;
 
+    /**
+     * <p>Creates a new {@link HttpField} with the given {@link HttpHeader},
+     * name string and value string.</p>
+     * <p>A {@code null} field value may be passed as parameter, and will
+     * be converted to the empty string.
+     * This allows the direct constructions of fields that have no value,
+     * and/or {@link HttpField} subclasses that override {@link #getValue()}.</p>
+     *
+     * @param header the {@link HttpHeader} referencing a well-known HTTP header name;
+     * may be {@code null} in case of an unknown or custom HTTP header name
+     * @param name the field name; if {@code null}, then {@link HttpHeader#asString()} is used
+     * @param value the field value; if {@code null}, the empty string will be used
+     */
     public HttpField(HttpHeader header, String name, String value)
     {
         _header = header;
         if (_header != null && name == null)
             _name = _header.asString();
         else
             _name = Objects.requireNonNull(name, "name");
-        _value = value;
+        _value = value != null ? value : "";
     }
 
+    /**
+     * <p>Creates a new {@link HttpField} with the given {@link HttpHeader}
+     * and value string.</p>
+     *
+     * @param header the non-{@code null} {@link HttpHeader} referencing a well-known HTTP header name
+     * @param value the field value; if {@code null}, the empty string will be used
+     */
     public HttpField(HttpHeader header, String value)
     {
         this(header, header.asString(), value);
     }
 
+    /**
+     * <p>Creates a new {@link HttpField} with the given {@link HttpHeader}
+     * and value.</p>
+     *
+     * @param header the non-{@code null} {@link HttpHeader} referencing a well-known HTTP header name
+     * @param value the field value; if {@code null}, the empty string will be used
+     */
     public HttpField(HttpHeader header, HttpHeaderValue value)
     {
-        this(header, header.asString(), value.asString());
+        this(header, header.asString(), value != null ? value.asString() : null);
     }
 
+    /**
+     * <p>Creates a new {@link HttpField} with the given name string
+     * and value string.</p>
+     *
+     * @param name the non-{@code null} field name
+     * @param value the field value; if {@code null}, the empty string will be used
+     */
     public HttpField(String name, String value)
     {
         this(HttpHeader.CACHE.get(name), name, value);
```

Comment:
What happens if `name` is non-null but `CACHE.get(name)` returns `null`? Shouldn't you add new values to the cache if they're missing?

Reply:
This is a cache to an enum. Since we can't add to the enum, we can't add to the cache.

Reply:
Fair enough, but won't users end up with a confusing error message or behavior if no match is found?